src/libexpr/primops.cc

```diff
@@ -564,22 +564,29 @@ static Bindings::iterator getAttr(
             funcName
         );
 
-        Pos aPos = *attrSet->pos;
-        if (aPos == noPos) {
+        if (attrSet->pos == nullptr) {
```

Comment:
This might be factored out a slight bit I guess. Something along the lines of:

```cpp
if (attrSet->pos && *attrSet->pos != noPos) {
  // Throw with the right location and the stack trace
} else {
  // Throw with the parent location
}
```

Reply:
Does this look OK to you now?